### PR TITLE
History functionality implemented

### PR DESCRIPTION
**Pull request by shobit1337**
- Completed History Page and its functionality.
- Create required utilities for history.
- Added History functionality to Video Page.

### CODE REVIEW

File: src/utils/history/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import axios from 'axios';
+
+const API_URL = process.env.REACT_APP_API_URL;
+
+export const getHistory = async (token) => {
+  try {
+    let { data } = await axios.get(`${API_URL}/user/history`, {
```
Comment (rutvikpumak) [inline src/utils/history/index.js:7]: make const

File: src/utils/history/index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    if (data.history) return data.history;
+    return [];
+  } catch (err) {
+    throw new Error('Failed to get History.');
+  }
+};
+
+export const setHistory = async (token, video) => {
+  try {
+    let { data } = await axios.post(
```
Comment (rutvikpumak) [inline src/utils/history/index.js:19]: make const

File: src/utils/history/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    return [];
+  } catch (err) {
+    console.log('Error: ', err);
+    throw new Error('Failed to add to history.');
+  }
+};
+
+export const deleteHistory = async (token, videoId) => {
+  try {
+    let { data } = await axios.delete(`${API_URL}/user/history/${videoId}`, {
```
Comment (rutvikpumak) [inline src/utils/history/index.js:36]: moake const

**Review by rutvikpumak — APPROVED** [review]
The code looks clean and structured.
Suggested a few changes and then it's good to merge.

File: src/pages/HistoryPage/HistoryPage.jsx [hunk 4]
```diff
@@ -1,7 +1,67 @@
...
+              <span
+                className='delete-video-btn cursor-pointer'
+                onClick={() => handleDeleteFromHistory(video._id)}>
+                ❌
+              </span>
+            </div>
+          ))
+        ) : (
+          <div className='text-secondary text-center text-md flex-grow'>
+            {' '}
```
Comment (manismk) [inline src/pages/HistoryPage/HistoryPage.jsx:58]: can remove these empty strings

**Review by manismk — APPROVED** [review]
Code is easy to understand and functionality works properly

File: src/utils/history/index.js (hunk 3 above)
Comment (shobit1337) [reply src/utils/history/index.js:36]: I'm following `let` at many places, Will raise a different PR for that. Thank you

**Pull request merged**